src/lib.rs

```diff
@@ -61,34 +64,74 @@ pub mod pallet {
     #[pallet::storage]
     pub type VMStorage<T> = StorageMap<_, Blake2_128Concat, Vec<u8>, Vec<u8>>;
 
+    /// Storage for multi-signature/signer requests.
+    #[pallet::storage]
+    pub type Multisigs<T: Config> = StorageDoubleMap<
+        _,
+        Twox64Concat,
+        T::AccountId,
+        Blake2_128Concat,
+        CallHash,
+        Multisig<BlockNumberFor<T>, T::AccountId, T::MaxSignatories>,
+    >;
+
     /// MoveVM pallet configuration trait
     #[pallet::config]
     pub trait Config: frame_system::Config {
+        /// The currency mechanism.
+        type Currency: Currency<Self::AccountId>
+            + ReservableCurrency<Self::AccountId>
+            + LockableCurrency<Self::AccountId>;
+
+        /// Maximum life time for requests.
+        #[pallet::constant]
+        type MaxRequestLifetime: Get<u32>;
+
+        /// Maximum number of signatories in multi-signer requests.
+        #[pallet::constant]
+        type MaxSignatories: Get<u32>;
```

Comment:
This would be a more direct name that reflects what this really represents:
```suggestion
        type MaxScriptSigners: Get<u32>;
```
`MaxSignatories` can mean more things, it could imply a much broader context

Reply:
Done